src/main/java/org/springframework/data/elasticsearch/core/MappingBuilder.java

```diff
@@ -360,6 +363,7 @@ private void addFieldMappingParameters(XContentBuilder builder, Object annotatio
 			searchAnalyzer = fieldAnnotation.searchAnalyzer();
 			normalizer = fieldAnnotation.normalizer();
 			copyTo = fieldAnnotation.copyTo();
+			ignoreAbove = fieldAnnotation.ignoreAbove();
```

Comment:
dd `Assert.isTrue(ignoreAbove > 0, "ignore_above must be a positive value");` to prevent illegal values, same in the `else if` block

Reply:
Hi, since the annotation defaults to -1, is this assertion necessary?

Reply:
I just checked, 0 is an allowed value, the question is, what do we do if someone configures `-7`? If we want to prevent that, we need the Assertion (but then on `>= 0`), but we could also just leave the Assert out and ignore the negative value. I'd prefer to have the assertion here instead of the user having invalid and ignored code.